### ISSUE TITLE
Scalability tests are failing since Sep 17th. 

### ISSUE THREAD

**Issue by gmarek** [kubernetes/kubernetes#14273]
cc @wojtek-t @quinton-hoole

**Comment by quinton-hoole** [comment]
Some of the recent errors you're seeing are:
Multiple e2e tests fail with "Namespace x is active" #13828 
... and possibly openshift/origin#4731

**Comment by gmarek** [comment]
I'm still seeing:
```
Density [Skipped] [Performance suite] should allow starting 30 pods per node

/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/density.go:410
Expected error:
    <*errors.errorString | 0xc212708530>: {
        s: "Number of reported pods changed: 2982 vs 3000",
    }
    Number of reported pods changed: 2982 vs 3000
not to have occurred
```

**Comment by gmarek** [comment]
It's still failing though with different error: 
```
03:45:50   Expected error:
03:45:50       <*errors.errorString | 0xc2212f2e90>: {
03:45:50           s: "gave up waiting for pod 'density-latency-pod-13a751c5-6114-11e5-9c0e-42010af01555-4' to be 'running' after 5m0s",
03:45:50       }
03:45:50       gave up waiting for pod 'density-latency-pod-13a751c5-6114-11e5-9c0e-42010af01555-4' to be 'running' after 5m0s
03:45:50   not to have occurred
```

I'm seeing in on my clusters as well. It's actually quite strange, as it seems that it depends on the number of scheduled pods: it reliably time outs on 100 node cluster, and it *kind of* succeeds on 3 node one. *Kind of* because e2e startup is just crazy:
```
Sep 22 13:57:21.323: INFO: 10% worst schedule latencies: [{density-latency-pod-dcc8b01a-6120-11e5-9545-7446a0bae05f-3 e2e-test-gmarek-minion-aljb 0}]
Sep 22 13:57:21.323: INFO: perc50: 0, perc90: 0, perc99: 0
Sep 22 13:57:21.323: INFO: 10% worst run-after-schedule latencies: [{density-latency-pod-dcc8b01a-6120-11e5-9545-7446a0bae05f-3 e2e-test-gmarek-minion-aljb 1s}]
Sep 22 13:57:21.323: INFO: perc50: 1s, perc90: 1s, perc99: 1s
Sep 22 13:57:21.323: INFO: 10% worst watch latencies: [{density-latency-pod-dcc8b01a-6120-11e5-9545-7446a0bae05f-3 e2e-test-gmarek-minion-aljb 1m31.864543958s}]
Sep 22 13:57:21.323: INFO: perc50: 1m31.66538035s, perc90: 1m31.864543958s, perc99: 1m31.864543958s
Sep 22 13:57:21.323: INFO: 10% worst scheduled-to-end total latencies: [{density-latency-pod-dcc8b01a-6120-11e5-9545-7446a0bae05f-3 e2e-test-gmarek-minion-aljb 1m32.864543958s}]
Sep 22 13:57:21.323: INFO: perc50: 1m32.66538035s, perc90: 1m32.864543958s, perc99: 1m32.864543958s
Sep 22 13:57:21.323: INFO: 10% worst e2e total latencies: [{density-latency-pod-dcc8b01a-6120-11e5-9545-7446a0bae05f-3 e2e-test-gmarek-minion-aljb 1m32.864543958s}]
Sep 22 13:57:21.323: INFO: perc50: 1m32.66538035s, perc90: 1m32.864543958s, perc99: 1m32.864543958s
```

Another thing is that it's triggered only for additional 'latency' pods that are being scheduled after 'main' density part, i.e. when the cluster is saturated. First 30 pods/node are starting without any hiccup - at least I didn't notice any.

It's even stranger because the Container seems to be running (as suggested by events), but the status is not being reported (my guess is that Kubelet for some reason skips the update):
```
Name:				density-latency-pod-dcc8b01a-6120-11e5-9545-7446a0bae05f-2
Namespace:			e2e-tests-density-x9db6
Image(s):			gcr.io/google_containers/pause:go
Node:				e2e-test-gmarek-minion-aljb/
Labels:				name=density-latency-pod-dcc8b01a-6120-11e5-9545-7446a0bae05f
Status:				Pending
Reason:				
Message:			
IP:				
Replication Controllers:	<none>
Containers:
  density-latency-pod-dcc8b01a-6120-11e5-9545-7446a0bae05f-2:
    Image:		gcr.io/google_containers/pause:go
    State:		Waiting
    Ready:		False
    Restart Count:	0
Events:
  FirstSeen				LastSeen			Count	From					SubobjectPath									Reason		Message
  Tue, 22 Sep 2015 13:55:48 +0200	Tue, 22 Sep 2015 13:55:48 +0200	1	{scheduler }														Scheduled	Successfully assigned density-latency-pod-dcc8b01a-6120-11e5-9545-7446a0bae05f-2 to e2e-test-gmarek-minion-aljb
  Tue, 22 Sep 2015 13:55:48 +0200	Tue, 22 Sep 2015 13:55:48 +0200	1	{kubelet e2e-test-gmarek-minion-aljb}	implicitly required container POD						Pulled		Container image "gcr.io/google_containers/pause:0.8.0" already present on machine
  Tue, 22 Sep 2015 13:55:48 +0200	Tue, 22 Sep 2015 13:55:48 +0200	1	{kubelet e2e-test-gmarek-minion-aljb}	implicitly required container POD						Created		Created with docker id c42d841df4c1
  Tue, 22 Sep 2015 13:55:49 +0200	Tue, 22 Sep 2015 13:55:49 +0200	1	{kubelet e2e-test-gmarek-minion-aljb}	implicitly required container POD						Started		Started with docker id c42d841df4c1
  Tue, 22 Sep 2015 13:55:49 +0200	Tue, 22 Sep 2015 13:55:49 +0200	1	{kubelet e2e-test-gmarek-minion-aljb}	spec.containers{density-latency-pod-dcc8b01a-6120-11e5-9545-7446a0bae05f-2}	Pulled		Container image "gcr.io/google_containers/pause:go" already present on machine
  Tue, 22 Sep 2015 13:55:49 +0200	Tue, 22 Sep 2015 13:55:49 +0200	1	{kubelet e2e-test-gmarek-minion-aljb}	spec.containers{density-latency-pod-dcc8b01a-6120-11e5-9545-7446a0bae05f-2}	Created		Created with docker id da00f7728836
  Tue, 22 Sep 2015 13:55:49 +0200	Tue, 22 Sep 2015 13:55:49 +0200	1	{kubelet e2e-test-gmarek-minion-aljb}	spec.containers{density-latency-pod-dcc8b01a-6120-11e5-9545-7446a0bae05f-2}	Started		Started with docker id da00f7728836
```

My guess is that it's something in Kubelet that prevents it from updating the status (cc @dchen1107 @yujuhong), but it's also possible it's on the watch side (cc @lavalamp @davidopp @wojtek-t)

**Comment by gmarek** [comment]
cc @smarterclayton @timothysc

**Comment by timothysc** [comment]
So there are several orthogonal issues being discussed at once here, so lets break them down: 

> Multiple e2e tests fail with "Namespace x is active" #13828

IMHO this is an annoyance that someone created when they started putting NS-deletes in front of the tests, the > the number of nodes + events the longer your NS delete (It is extremely slow) /cc @derekwaynecarr 

> "Number of reported pods changed: 2982 vs 3000"

Typically we've seen this if pods start failing, this can sometimes be caused by kubelet issues, or pod failures.  

> "gave up waiting for pod 'density-latency-pod-13a751c5-6114-11e5-9c0e-42010af01555-4' to be 'running' after 5m0s"

I'd like to see the complete logs if possible on the different runs.

**Comment by gmarek** [comment]
@timothysc I can provide you with logs, but it fails reliably for me on the head, when run on GCE. Actually I'm curious if it also happens with other setups. I suggest moving discussion to slack scalability channel.

As for active namespaces - noone is actually deleting NSs in front of the tests. It was a really, really, really bad name of a function.

Inability to start 30 pods/node was indeed most likely caused by 'Node not ready' problem, so it's most likely fixed by @yujuhong revert.

**Comment by timothysc** [comment]
For completeness: 

We see a similar issue on test cleanup timing out.  
http://fpaste.org/270182/44293929/

**Comment by yujuhong** [comment]
> My guess is that it's something in Kubelet that prevents it from updating the status (cc @dchen1107 @yujuhong), but it's also possible it's on the watch side (cc @lavalamp @davidopp @wojtek-t)

Disclaimer: I haven't looked into the failures yet.

My past experience with the pod startup latency is that as the #pods increases, the apiserver QPS limit becomes the major bottleneck since we started triggering pod worker independently (#13003). The pods would be ready long before kubelet can send out status updates. Any change that would increase #requests kubelet sends to the apiserver may increase the pod startup latency.

@gmarek, have you increased the timeout to see if kubelet *eventually* updates the status for all pods?

**Comment by lavalamp** [comment]
I will look through the logs of a recent run and see if anything stands out.

**Comment by lavalamp** [comment]
Scheduler log has a lot of lines like:

```
Combined requested resources 3300 from existing pods exceeds capacity 2000 on node e2e-scalability-minion-4juo
```

Which is troubling-- are we scheduling stuff in some biased manner? Existing pods on a node shouldn't have resource sums higher than the capacity?

**Comment by lavalamp** [comment]
Filed #14384 - currently don't think it's the root cause.

**Comment by lavalamp** [comment]
BTW, I'm examining logs from http://kubekins.dls.corp.google.com/job/kubernetes-e2e-gce-scalability/2507/

The specific failure is `gave up waiting for pod 'density-latency-pod-13a751c5-6114-11e5-9c0e-42010af01555-4' to be 'running' after 5m0s`

That pod is on node `e2e-scalability-minion-rrnh`. Here's some logs in the midst of the problematic behavior.

```
I0922 10:36:13.884852       5 handlers.go:143] GET /api/v1/pods?fieldSelector=spec.nodeName%3De2e-scalability-minion-rrnh: (18.485022573s) 200 [[kubelet/v1.1.0 (linux/amd64) kubernetes/09cf38e] 10.240.84.162:39917]
...
I0922 10:36:14.667254       5 handlers.go:143] GET /api/v1/watch/pods?fieldSelector=spec.nodeName%3De2e-scalability-minion-rrnh&resourceVersion=338101: (709.10701ms) 0 [[kubelet/v1.1.0 (linux/amd64) kubernetes/09cf38e] 
...
I0922 10:36:27.254781       5 handlers.go:143] GET /api/v1/pods?fieldSelector=spec.nodeName%3De2e-scalability-minion-rrnh: (11.524191846s) 200 [[kubelet/v1.1.0 (linux/amd64) kubernetes/09cf38e] 10.240.84.162:39917]
...
goroutine 1337888 [running]:
k8s.io/kubernetes/pkg/httplog.(*respLogger).WriteHeader(0xc2b1e7a480, 0x1f4)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/httplog/log.go:188 +0x9a
k8s.io/kubernetes/pkg/apiserver.writeJSON(0x1f4, 0x7f4c2d21b238, 0xc2080f7120, 0x7f4c2d21b9d0, 0xc2e6266a00, 0x7f4c2d069698, 0xc2b1e7a480, 0x1)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/apiserver/apiserver.go:399 +0x2b7
k8s.io/kubernetes/pkg/apiserver.errorJSON(0x7f4c2d215f40, 0xc2e3f47d00, 0x7f4c2d21b238, 0xc2080f7120, 0x7f4c2d069698, 0xc2b1e7a480, 0xc2c6ad8760)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/apiserver/apiserver.go:406 +0xab
k8s.io/kubernetes/pkg/apiserver.func·027(0xc211459ad0, 0xc2b1e7a540)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/apiserver/resthandler.go:256 +0xc2a
github.com/emicklei/go-restful.func·009(0xc211459ad0, 0xc2b1e7a540)
	/go/src/k8s.io/kubernetes/Godeps/_workspace/src/github.com/emicklei/go-restful/container.go:244 +0x41
github.com/emicklei/go-restful.(*FilterChain).ProcessFilter(0xc211459b60, 0xc211459ad0, 0xc2b1e7a540)
	/go/src/k8s.io/kubernetes/Godeps/_workspace/src/github.com/emicklei/go-restful/filter.go:21 +0xa2
k8s.io/kubernetes/pkg/apiserver.func·003(0xc211459ad0, 0xc2b1e7a540, 0xc211459b60)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/apiserver/apiserver.go:61 +0x8b
github.com/emicklei/go-restful.(*FilterChain).ProcessFilter(0xc211459b60, 0xc211459ad0, 0xc2b1e7a540)
	/go/src/k8s.io/kubernetes/Godeps/_workspace/src/github.com/emicklei/go-restful/filter.go:19 +0x84
github.com/emicklei/go-restful.(*Container).dispatch(0xc2082a8090, 0x7f4c2d069698, 0xc2b1e7a480, 0xc213644b60)
	/go/src/k8s.io/kubernetes/Godeps/_workspace/src/github.com/emicklei/go-restful/container.go:246 +0xee3
github.com/emicklei/go-restful.*Container.(github.com/emicklei/go-restful.dispatch)·fm(0x7f4c2d069698, 0xc2b1e7a480, 0xc213644b60)
	/go/src/k8s.io/kubernetes/Godeps/_workspa [[kubelet/v1.1.0 (linux/amd64) kubernetes/09cf38e] 10.240.252.125:54303]
E0922 10:36:31.057767       5 errors.go:62] apiserver received an error that is not an unversioned.Status: too old resource version: 338159 (341885)
I0922 10:36:31.057953       5 handlers.go:143] GET /api/v1/watch/pods?fieldSelector=spec.nodeName%3De2e-scalability-minion-rrnh&resourceVersion=339493: (3.717997659s) 500
```

Note that the stack trace and the error lines don't correspond exactly, because threading and many similar things fail simultaneously.

Conclusions:
* A problem here is list/watch behavior. There's two examples above. In the first, the list took 18 (!) seconds, and the watch ended (unknown reason) after only 700ms. In the second, the list took 11s, and when the watch was attempted only 4 seconds later, it was already too late, the resource version was out of date.
* This causes kubelets to basically poll. It's possible that this interferes with them posting pod status updates, in which case this problem could be a root cause.
* This needs to be fixed whether or not it's the root cause here, I will file a separate issue. We need to start catching watch problems at the integration test level.

**Comment by yujuhong** [comment]
It's probably caused by #12894. kubelet only sends pod status update when there is any change (status diff). #12894 added a new field "LastProbeTime", which changes whenever kubelet generates a status. This could cause significantly more status updates being sent to the apiserver. 

e.g., pod status update for a single pod over the period of the time. You can see that the status was updated ~10 to 30 seconds.
```
I0922 19:13:29.074799   16658 manager.go:223] Status for pod "density3000-c3f4b6be-615c-11e5-b85d-42010af01555-2y1h9_e2e-tests-density-b0a8z" updated successfully
I0922 19:13:51.186106   16658 manager.go:223] Status for pod "density3000-c3f4b6be-615c-11e5-b85d-42010af01555-2y1h9_e2e-tests-density-b0a8z" updated successfully
I0922 19:14:04.031949   16658 manager.go:223] Status for pod "density3000-c3f4b6be-615c-11e5-b85d-42010af01555-2y1h9_e2e-tests-density-b0a8z" updated successfully
I0922 19:14:25.107090   16658 manager.go:223] Status for pod "density3000-c3f4b6be-615c-11e5-b85d-42010af01555-2y1h9_e2e-tests-density-b0a8z" updated successfully
I0922 19:14:56.027125   16658 manager.go:223] Status for pod "density3000-c3f4b6be-615c-11e5-b85d-42010af01555-2y1h9_e2e-tests-density-b0a8z" updated successfully
I0922 19:15:07.445418   16658 manager.go:223] Status for pod "density3000-c3f4b6be-615c-11e5-b85d-42010af01555-2y1h9_e2e-tests-density-b0a8z" updated successfully
I0922 19:15:10.929029   16658 manager.go:223] Status for pod "density3000-c3f4b6be-615c-11e5-b85d-42010af01555-2y1h9_e2e-tests-density-b0a8z" updated successfully
I0922 19:15:42.027571   16658 manager.go:223] Status for pod "density3000-c3f4b6be-615c-11e5-b85d-42010af01555-2y1h9_e2e-tests-density-b0a8z" updated successfully
I0922 19:16:24.071344   16658 manager.go:223] Status for pod "density3000-c3f4b6be-615c-11e5-b85d-42010af01555-2y1h9_e2e-tests-density-b0a8z" updated successfully
I0922 19:16:33.516615   16658 manager.go:223] Status for pod "density3000-c3f4b6be-615c-11e5-b85d-42010af01555-2y1h9_e2e-tests-density-b0a8z" updated successfully
I0922 19:17:01.688647   16658 manager.go:223] Status for pod "density3000-c3f4b6be-615c-11e5-b85d-42010af01555-2y1h9_e2e-tests-density-b0a8z" updated successfully
I0922 19:17:20.511306   16658 manager.go:223] Status for pod "density3000-c3f4b6be-615c-11e5-b85d-42010af01555-2y1h9_e2e-tests-density-b0a8z" updated successfully
I0922 19:17:49.809671   16658 manager.go:223] Status for pod "density3000-c3f4b6be-615c-11e5-b85d-42010af01555-2y1h9_e2e-tests-density-b0a8z" updated successfully
I0922 19:18:19.044876   16658 manager.go:223] Status for pod "density3000-c3f4b6be-615c-11e5-b85d-42010af01555-2y1h9_e2e-tests-density-b0a8z" updated successfully
I0922 19:18:23.388399   16658 manager.go:223] Status for pod "density3000-c3f4b6be-615c-11e5-b85d-42010af01555-2y1h9_e2e-tests-density-b0a8z" updated successfully
I0922 19:18:57.245296   16658 manager.go:223] Status for pod "density3000-c3f4b6be-615c-11e5-b85d-42010af01555-2y1h9_e2e-tests-density-b0a8z" updated successfully
I0922 19:19:20.281819   16658 manager.go:223] Status for pod "density3000-c3f4b6be-615c-11e5-b85d-42010af01555-2y1h9_e2e-tests-density-b0a8z" updated successfully
I0922 19:19:51.462922   16658 manager.go:223] Status for pod "density3000-c3f4b6be-615c-11e5-b85d-42010af01555-2y1h9_e2e-tests-density-b0a8z" updated successfully
I0922 19:20:18.284142   16658 manager.go:223] Status for pod "density3000-c3f4b6be-615c-11e5-b85d-42010af01555-2y1h9_e2e-tests-density-b0a8z" updated successfully
I0922 19:20:28.096845   16658 manager.go:223] Status for pod "density3000-c3f4b6be-615c-11e5-b85d-42010af01555-2y1h9_e2e-tests-density-b0a8z" updated successfully
I0922 19:20:33.863465   16658 manager.go:223] Status for pod "density3000-c3f4b6be-615c-11e5-b85d-42010af01555-2y1h9_e2e-tests-density-b0a8z" updated successfully
I0922 19:21:00.072203   16658 manager.go:223] Status for pod "density3000-c3f4b6be-615c-11e5-b85d-42010af01555-2y1h9_e2e-tests-density-b0a8z" updated successfully
I0922 19:21:29.015362   16658 manager.go:223] Status for pod "density3000-c3f4b6be-615c-11e5-b85d-42010af01555-2y1h9_e2e-tests-density-b0a8z" updated successfully
I0922 19:21:37.880921   16658 manager.go:223] Status for pod "density3000-c3f4b6be-615c-11e5-b85d-42010af01555-2y1h9_e2e-tests-density-b0a8z" updated successfully
I0922 19:22:02.915922   16658 manager.go:223] Status for pod "density3000-c3f4b6be-615c-11e5-b85d-42010af01555-2y1h9_e2e-tests-density-b0a8z" updated successfully
I0922 19:22:41.111139   16658 manager.go:223] Status for pod "density3000-c3f4b6be-615c-11e5-b85d-42010af01555-2y1h9_e2e-tests-density-b0a8z" updated successfully
I0922 19:22:43.558756   16658 manager.go:223] Status for pod "density3000-c3f4b6be-615c-11e5-b85d-42010af01555-2y1h9_e2e-tests-density-b0a8z" updated successfully
I0922 19:23:13.136956   16658 manager.go:223] Status for pod "density3000-c3f4b6be-615c-11e5-b85d-42010af01555-2y1h9_e2e-tests-density-b0a8z" updated successfully
I0922 19:23:46.059183   16658 manager.go:223] Status for pod "density3000-c3f4b6be-615c-11e5-b85d-42010af01555-2y1h9_e2e-tests-density-b0a8z" updated successfully
I0922 19:23:54.381561   16658 manager.go:223] Status for pod "density3000-c3f4b6be-615c-11e5-b85d-42010af01555-2y1h9_e2e-tests-density-b0a8z" updated successfully
I0922 19:24:45.978286   16658 manager.go:223] Status for pod "density3000-c3f4b6be-615c-11e5-b85d-42010af01555-2y1h9_e2e-tests-density-b0a8z" updated successfully
I0922 19:25:19.182797   16658 manager.go:223] Status for pod "density3000-c3f4b6be-615c-11e5-b85d-42010af01555-2y1h9_e2e-tests-density-b0a8z" updated successfully
I0922 19:25:30.239748   16658 manager.go:223] Status for pod "density3000-c3f4b6be-615c-11e5-b85d-42010af01555-2y1h9_e2e-tests-density-b0a8z" updated successfully
I0922 19:26:00.684613   16658 manager.go:223] Status for pod "density3000-c3f4b6be-615c-11e5-b85d-42010af01555-2y1h9_e2e-tests-density-b0a8z" updated successfully
I0922 19:26:36.039905   16658 manager.go:223] Status for pod "density3000-c3f4b6be-615c-11e5-b85d-42010af01555-2y1h9_e2e-tests-density-b0a8z" updated successfully
I0922 19:27:04.624735   16658 manager.go:223] Status for pod "density3000-c3f4b6be-615c-11e5-b85d-42010af01555-2y1h9_e2e-tests-density-b0a8z" updated successfully
I0922 19:27:29.233732   16658 manager.go:223] Status for pod "density3000-c3f4b6be-615c-11e5-b85d-42010af01555-2y1h9_e2e-tests-density-b0a8z" updated successfully
I0922 19:27:56.099400   16658 manager.go:223] Status for pod "density3000-c3f4b6be-615c-11e5-b85d-42010af01555-2y1h9_e2e-tests-density-b0a8z" updated successfully
I0922 19:28:29.565595   16658 manager.go:223] Status for pod "density3000-c3f4b6be-615c-11e5-b85d-42010af01555-2y1h9_e2e-tests-density-b0a8z" updated successfully
I0922 19:29:02.739256   16658 manager.go:223] Status for pod "density3000-c3f4b6be-615c-11e5-b85d-42010af01555-2y1h9_e2e-tests-density-b0a8z" updated successfully
I0922 19:29:12.202317   16658 manager.go:223] Status for pod "density3000-c3f4b6be-615c-11e5-b85d-42010af01555-2y1h9_e2e-tests-density-b0a8z" updated successfully
I0922 19:29:36.494084   16658 manager.go:223] Status for pod "density3000-c3f4b6be-615c-11e5-b85d-42010af01555-2y1h9_e2e-tests-density-b0a8z" updated successfully
I0922 19:30:09.085855   16658 manager.go:223] Status for pod "density3000-c3f4b6be-615c-11e5-b85d-42010af01555-2y1h9_e2e-tests-density-b0a8z" updated successfully
I0922 19:30:41.116653   16658 manager.go:223] Status for pod "density3000-c3f4b6be-615c-11e5-b85d-42010af01555-2y1h9_e2e-tests-density-b0a8z" updated successfully
I0922 19:31:03.392930   16658 manager.go:223] Status for pod "density3000-c3f4b6be-615c-11e5-b85d-42010af01555-2y1h9_e2e-tests-density-b0a8z" updated successfully
I0922 19:31:17.831217   16658 manager.go:223] Status for pod "density3000-c3f4b6be-615c-11e5-b85d-42010af01555-2y1h9_e2e-tests-density-b0a8z" updated successfully
I0922 19:32:12.663789   16658 manager.go:223] Status for pod "density3000-c3f4b6be-615c-11e5-b85d-42010af01555-2y1h9_e2e-tests-density-b0a8z" updated successfully
I0922 19:32:22.400875   16658 manager.go:223] Status for pod "density3000-c3f4b6be-615c-11e5-b85d-42010af01555-2y1h9_e2e-tests-density-b0a8z" updated successfully
I0922 19:33:09.218379   16658 manager.go:223] Status for pod "density3000-c3f4b6be-615c-11e5-b85d-42010af01555-2y1h9_e2e-tests-density-b0a8z" updated successfully
I0922 19:33:20.133960   16658 manager.go:223] Status for pod "density3000-c3f4b6be-615c-11e5-b85d-42010af01555-2y1h9_e2e-tests-density-b0a8z" updated successfully
I0922 19:33:53.506832   16658 manager.go:223] Status for pod "density3000-c3f4b6be-615c-11e5-b85d-42010af01555-2y1h9_e2e-tests-density-b0a8z" updated successfully
I0922 19:34:28.099576   16658 manager.go:223] Status for pod "density3000-c3f4b6be-615c-11e5-b85d-42010af01555-2y1h9_e2e-tests-density-b0a8z" updated successfully
I0922 19:34:38.551262   16658 manager.go:223] Status for pod "density3000-c3f4b6be-615c-11e5-b85d-42010af01555-2y1h9_e2e-tests-density-b0a8z" updated successfully
I0922 19:35:06.433594   16658 manager.go:223] Status for pod "density3000-c3f4b6be-615c-11e5-b85d-42010af01555-2y1h9_e2e-tests-density-b0a8z" updated successfully
I0922 19:36:11.269109   16658 manager.go:223] Status for pod "density3000-c3f4b6be-615c-11e5-b85d-42010af01555-2y1h9_e2e-tests-density-b0a8z" updated successfully
I0922 19:36:12.850897   16658 manager.go:223] Status for pod "density3000-c3f4b6be-615c-11e5-b85d-42010af01555-2y1h9_e2e-tests-density-b0a8z" updated successfully
I0922 19:36:42.396782   16658 manager.go:223] Status for pod "density3000-c3f4b6be-615c-11e5-b85d-42010af01555-2y1h9_e2e-tests-density-b0a8z" updated successfully
I0922 19:36:43.653108   16658 manager.go:223] Status for pod "density3000-c3f4b6be-615c-11e5-b85d-42010af01555-2y1h9_e2e-tests-density-b0a8z" updated successfully
I0922 19:37:19.058685   16658 manager.go:223] Status for pod "density3000-c3f4b6be-615c-11e5-b85d-42010af01555-2y1h9_e2e-tests-density-b0a8z" updated successfully
I0922 19:37:53.057503   16658 manager.go:223] Status for pod "density3000-c3f4b6be-615c-11e5-b85d-42010af01555-2y1h9_e2e-tests-density-b0a8z" updated successfully
I0922 19:38:20.011815   16658 manager.go:223] Status for pod "density3000-c3f4b6be-615c-11e5-b85d-42010af01555-2y1h9_e2e-tests-density-b0a8z" updated successfully
I0922 19:38:50.617293   16658 manager.go:223] Status for pod "density3000-c3f4b6be-615c-11e5-b85d-42010af01555-2y1h9_e2e-tests-density-b0a8z" updated successfully
I0922 19:39:10.544237   16658 manager.go:223] Status for pod "density3000-c3f4b6be-615c-11e5-b85d-42010af01555-2y1h9_e2e-tests-density-b0a8z" updated successfully
I0922 19:39:52.112816   16658 manager.go:223] Status for pod "density3000-c3f4b6be-615c-11e5-b85d-42010af01555-2y1h9_e2e-tests-density-b0a8z" updated successfully
I0922 19:40:32.162664   16658 manager.go:223] Status for pod "density3000-c3f4b6be-615c-11e5-b85d-42010af01555-2y1h9_e2e-tests-density-b0a8z" updated successfully
I0922 19:40:35.121094   16658 manager.go:223] Status for pod "density3000-c3f4b6be-615c-11e5-b85d-42010af01555-2y1h9_e2e-tests-density-b0a8z" updated successfully
I0922 19:41:35.670167   16658 manager.go:223] Status for pod "density3000-c3f4b6be-615c-11e5-b85d-42010af01555-2y1h9_e2e-tests-density-b0a8z" updated successfully
I0922 19:41:37.037979   16658 manager.go:223] Status for pod "density3000-c3f4b6be-615c-11e5-b85d-42010af01555-2y1h9_e2e-tests-density-b0a8z" updated successfully
I0922 19:41:39.129615   16658 manager.go:223] Status for pod "density3000-c3f4b6be-615c-11e5-b85d-42010af01555-2y1h9_e2e-tests-density-b0a8z" updated successfully
I0922 19:41:59.893938   16658 manager.go:223] Status for pod "density3000-c3f4b6be-615c-11e5-b85d-42010af01555-2y1h9_e2e-tests-density-b0a8z" updated successfully
I0922 19:42:36.821537   16658 manager.go:223] Status for pod "density3000-c3f4b6be-615c-11e5-b85d-42010af01555-2y1h9_e2e-tests-density-b0a8z" updated successfully
I0922 19:43:19.690384   16658 manager.go:223] Status for pod "density3000-c3f4b6be-615c-11e5-b85d-42010af01555-2y1h9_e2e-tests-density-b0a8z" updated successfully
I0922 19:43:41.196169   16658 manager.go:223] Status for pod "density3000-c3f4b6be-615c-11e5-b85d-42010af01555-2y1h9_e2e-tests-density-b0a8z" updated successfully
I0922 19:44:15.990440   16658 manager.go:223] Status for pod "density3000-c3f4b6be-615c-11e5-b85d-42010af01555-2y1h9_e2e-tests-density-b0a8z" updated successfully
I0922 19:44:42.611470   16658 manager.go:223] Status for pod "density3000-c3f4b6be-615c-11e5-b85d-42010af01555-2y1h9_e2e-tests-density-b0a8z" updated successfully
I0922 19:45:42.832205   16658 manager.go:223] Status for pod "density3000-c3f4b6be-615c-11e5-b85d-42010af01555-2y1h9_e2e-tests-density-b0a8z" updated successfully
I0922 19:45:44.764852   16658 manager.go:223] Status for pod "density3000-c3f4b6be-615c-11e5-b85d-42010af01555-2y1h9_e2e-tests-density-b0a8z" updated successfully
I0922 19:46:07.039022   16658 manager.go:223] Status for pod "density3000-c3f4b6be-615c-11e5-b85d-42010af01555-2y1h9_e2e-tests-density-b0a8z" updated successfully
I0922 19:46:37.233027   16658 manager.go:223] Status for pod "density3000-c3f4b6be-615c-11e5-b85d-42010af01555-2y1h9_e2e-tests-density-b0a8z" updated successfully
I0922 19:46:38.076356   16658 manager.go:223] Status for pod "density3000-c3f4b6be-615c-11e5-b85d-42010af01555-2y1h9_e2e-tests-density-b0a8z" updated successfully
I0922 19:47:19.969555   16658 manager.go:223] Status for pod "density3000-c3f4b6be-615c-11e5-b85d-42010af01555-2y1h9_e2e-tests-density-b0a8z" updated successfully
I0922 19:47:44.326752   16658 manager.go:223] Status for pod "density3000-c3f4b6be-615c-11e5-b85d-42010af01555-2y1h9_e2e-tests-density-b0a8z" updated successfully
I0922 19:47:55.092897   16658 manager.go:223] Status for pod "density3000-c3f4b6be-615c-11e5-b85d-42010af01555-2y1h9_e2e-tests-density-b0a8z" updated successfully
I0922 19:48:46.166302   16658 manager.go:223] Status for pod "density3000-c3f4b6be-615c-11e5-b85d-42010af01555-2y1h9_e2e-tests-density-b0a8z" updated successfully
I0922 19:48:47.308732   16658 manager.go:223] Status for pod "density3000-c3f4b6be-615c-11e5-b85d-42010af01555-2y1h9_e2e-tests-density-b0a8z" updated successfully
I0922 19:49:28.985278   16658 manager.go:223] Status for pod "density3000-c3f4b6be-615c-11e5-b85d-42010af01555-2y1h9_e2e-tests-density-b0a8z" updated successfully
I0922 19:49:53.595885   16658 manager.go:223] Status for pod "density3000-c3f4b6be-615c-11e5-b85d-42010af01555-2y1h9_e2e-tests-density-b0a8z" updated successfully
I0922 19:50:56.834849   16658 manager.go:223] Status for pod "density3000-c3f4b6be-615c-11e5-b85d-42010af01555-2y1h9_e2e-tests-density-b0a8z" updated successfully
I0922 19:50:58.414255   16658 manager.go:223] Status for pod "density3000-c3f4b6be-615c-11e5-b85d-42010af01555-2y1h9_e2e-tests-density-b0a8z" updated successfully
I0922 19:51:01.797375   16658 manager.go:223] Status for pod "density3000-c3f4b6be-615c-11e5-b85d-42010af01555-2y1h9_e2e-tests-density-b0a8z" updated successfully
I0922 19:51:28.085674   16658 manager.go:223] Status for pod "density3000-c3f4b6be-615c-11e5-b85d-42010af01555-2y1h9_e2e-tests-density-b0a8z" updated successfully
I0922 19:52:32.782354   16658 manager.go:223] Status for pod "density3000-c3f4b6be-615c-11e5-b85d-42010af01555-2y1h9_e2e-tests-density-b0a8z" updated successfully
I0922 19:52:34.229331   16658 manager.go:223] Status for pod "density3000-c3f4b6be-615c-11e5-b85d-42010af01555-2y1h9_e2e-tests-density-b0a8z" updated successfully
I0922 19:53:02.455412   16658 manager.go:223] Status for pod "density3000-c3f4b6be-615c-11e5-b85d-42010af01555-2y1h9_e2e-tests-density-b0a8z" updated successfully

```

/cc @nikhiljindal

**Comment by lavalamp** [comment]
@yujuhong that makes a bunch of sense. This is a good candidate for a root cause.

**Comment by lavalamp** [comment]
IRL discussion: @yujuhong is going to make a PR to make kubelet stop setting lastProbeTime to get the tests green. We can figure out what to do about this (and whether to keep it in the API) in parallel.

@bgrant0607 @nikhiljindal - do you guys have objections to this plan?

**Comment by lavalamp** [comment]
Thanks for #14390, @yujuhong-- Since no good deed goes unpunished, I gave you #14391 to catch the general class of errors, since I think this has happened before.

**Comment by gmarek** [comment]
@yujuhong - It's probably too late for this answer, but yes - Kubelet eventually reports those pods as running.

**Comment by gmarek** [comment]
I went through Jenkins logs and I found a run with "good" -the latest I found was scalability run 2465, which was started ~6pm 18th. It may be an extreme luck, but I haven't seen a 'fast run' since. I'll start looking for culprit from weekend... (which can mean it can be #12894, which was merged on 19th, but for some reason revert is not enough...)

**Comment by yujuhong** [comment]
> (which can mean it can be #12894, which was merged on 19th, but for some reason revert is not enough...)

@gmarek, could you grep kubelet log and see if the pod status updates are back to normal after your revert?

**Comment by gmarek** [comment]
I can't - my cluster is already down. Plus I don't really know what 'normal' means:)

**Comment by yujuhong** [comment]
> I can't - my cluster is already down. Plus I don't really know what 'normal' means:)

Too bad. "Normal" means that it should not send any update if there is no change :)

**Comment by wojtek-t** [comment]
I hope that it's finally fixed now (although latency metrics seems a bit worse than they used to be).

**Comment by gmarek** [comment]
It seems that the revert was enough to unbreak the tests, and latency I observed in test clusters was somehow skewed.

**Comment by gmarek** [comment]
We add number of pods times something to the resource usage. @davidopp

**Comment by brendandburns** [comment]
@gmarek can we close this now that #14554 is figured out and the remainder is fixed.

**Comment by gmarek** [comment]
Yup.
